File: src/NovelRT.Java/src/main/java/com/github/novelrt/WorldObject.java
package com.github.novelrt;

import com.github.novelrt.internal.HandleObject;
import com.github.novelrt.maths.Transform;

public class WorldObject extends HandleObject {
  protected WorldObject(long handle, boolean isOwned) {
    super(handle, isOwned, WorldObject::deleteWorldObject);
  }

  public static native void deleteWorldObject(long handle);

  public native Transform getTransform();

  public native int getLayer();
  public native void setLayer(int layer);

  public native boolean isActive();
  public native void setActive(boolean active);
}

File: src/NovelRT.Java/core/src/main/kotlin/com/github/novelrt/WorldObject.kt
package com.github.novelrt

import com.github.novelrt.fumocement.DisposalMethod
import com.github.novelrt.fumocement.HandleDeleter
import com.github.novelrt.interop.KotlinNativeObject

abstract class WorldObject internal constructor(
    handle: Long,
    isOwned: Boolean = true,
    deleter: HandleDeleter,
    disposalMethod: DisposalMethod = DisposalMethod.GARBAGE_COLLECTED
) : KotlinNativeObject(handle, isOwned, deleter, disposalMethod) {
    abstract val transform: Transform
    abstract var layer: Int
    abstract var active: Boolean
}
